### PR TITLE
Add dry-plan

### PR DESCRIPTION
**Pull request by grieve54706**
Add two APIs for the dry plan

One API only rewrites SQL with manifest
```
POST /v2/ibis/dry-plan
{
  "manifestStr": "eyJjYXRhbG9nIjoibXlfY2F0YWxvZyIsInN......",
  "sql": "SELECT orderkey, order_cust_key FROM \"Orders\" LIMIT 1"
}
```
```
WITH
    "Orders" AS (
        SELECT
            "Orders"."orderkey" AS "orderkey",
            "Orders"."custkey" AS "custkey",
            "Orders"."order_cust_key" AS "order_cust_key"
        FROM
            (
                SELECT
                    "Orders"."orderkey" AS "orderkey",
                    "Orders"."custkey" AS "custkey",
                    "Orders"."order_cust_key" AS "order_cust_key"
                FROM
                    (
                        SELECT
                            o_orderkey AS "orderkey",
                            o_custkey AS "custkey",
                            CONCAT (o_orderkey, '_', o_custkey) AS "order_cust_key"
                        FROM
                            (
                                SELECT
                                    o_orderkey,
                                    o_custkey
                                FROM
                                    public.orders
                            ) AS "Orders"
                    ) AS "Orders"
            ) AS "Orders"
    )
SELECT
    orderkey,
    order_cust_key
FROM
    "Orders"
LIMIT
    1
```

Another API will rewrite SQL with manifest and transpile by data source dialect.
```
POST /v2/ibis/{data_source}/dry-plan
{
  "manifestStr": "eyJjYXRhbG9nIjoibXlfY2F0YWxvZyIsInN......",
  "sql": "SELECT orderkey, order_cust_key FROM \"Orders\" LIMIT 1"
}
```
```
WITH
    "Orders" AS (
        SELECT
            "Orders"."orderkey" AS "orderkey",
            "Orders"."custkey" AS "custkey",
            "Orders"."order_cust_key" AS "order_cust_key"
        FROM
            (
                SELECT
                    "Orders"."orderkey" AS "orderkey",
                    "Orders"."custkey" AS "custkey",
                    "Orders"."order_cust_key" AS "order_cust_key"
                FROM
                    (
                        SELECT
                            o_orderkey AS "orderkey",
                            o_custkey AS "custkey",
                            CONCAT (o_orderkey, '_', o_custkey) AS "order_cust_key"
                        FROM
                            (
                                SELECT
                                    o_orderkey,
                                    o_custkey
                                FROM
                                    public.orders
                            ) AS "Orders"
                    ) AS "Orders"
            ) AS "Orders"
    )
SELECT
    orderkey,
    order_cust_key
FROM
    "Orders"
LIMIT
    1
```

### CODE REVIEW

**Comment by onlyjackfrost** [comment]
@grieve54706  Thanks for implementing these two APIs! 
Just one question, 
Is the SQL from the `POST /v2/ibis/dry-plan` API Trino-SQL? while the other is the native SQL that can be executed on each data source.

**Comment by grieve54706** [comment]
> @grieve54706 Thanks for implementing these two APIs! Just one question, Is the SQL from the `POST /v2/ibis/dry-plan` API Trino-SQL? while the other is the native SQL that can be executed on each data source.

The API will return Trino SQL now. But in the future, it will be Datafusion SQL when integrating Rust. They are not for each data source.

File: ibis-server/app/mdl/rewriter.py [hunk 1]
```diff
@@ -1,21 +1,50 @@
...
+        except httpx.ConnectError as e:
+            raise ConnectionError(f"Can not connect to Wren Engine: {e}")
+
+    def transpile(self, rewritten_sql: str) -> str:
+        transpiled_sql = sqlglot.transpile(
+            rewritten_sql, read="trino", write=self.data_source.name
+        )[0]
+        transpiled_sql = sqlglot.parse_one(
+            transpiled_sql, dialect=self.data_source.name
+        ).sql(self.data_source.name)
```
Comment (goldmedal) [inline ibis-server/app/mdl/rewriter.py:48]: Curiously, why we need this line? I do some tests, then I found the result of `sqlglot.parse_one().sql()` is equal to `sqlglot.transpile`? There're any additional check in this api?

File: ibis-server/tests/routers/test_ibis.py
Comment (goldmedal) [inline ibis-server/tests/routers/test_ibis.py:1]: Why the file called `test_ibis.py`? I think we can call it `test_dry_plan.py`?

**Review by goldmedal — APPROVED** [review]
Thanks @grieve54706 worked on this. LGTM. Just have some minor questions.

File: ibis-server/app/mdl/rewriter.py (hunk 1 above)
Comment (grieve54706) [reply ibis-server/app/mdl/rewriter.py:48]: Weird, I found a problem if only using `sqlglot.transpile` will get a SQL with invalid syntax `'` like `'WITH ABC ....'`. But now I re-test again, and it is gone.
I will remove this code. Thanks.

File: ibis-server/tests/routers/test_ibis.py
Comment (grieve54706) [reply ibis-server/tests/routers/test_ibis.py:1]: Make the test include APIs without the data source in `/v2/ibis`.

Comment (goldmedal) [reply ibis-server/tests/routers/test_ibis.py:1]: I see. Thanks

**Pull request merged**